torch/csrc/Exceptions.h

```diff
@@ -2,38 +2,57 @@
 
 #include <exception>
 #include <string>
+#include <memory>
+#include <queue>
 
 #include <torch/csrc/THP_export.h>
 #include <torch/csrc/utils/auto_gil.h>
 #include <torch/csrc/WindowsTorchApiMacro.h>
+#include "c10/util/StringUtil.h"
+#include "c10/util/Exception.h"
+
+/// NOTE [ Conversion Cpp Python Warning ]
+/// Python warning semantic is different from the cpp one in that
+/// they can raise errors. To handle this case which requires
```

Comment:
I'm not sure this first sentence makes sense to anyone who didn't participate in the design discussion. An example would make it a lot clearer, I think.

Reply:
Basically, walk the reader through the case when a warning handler raises an error, while the function itself is throwing an exception. (That diagram we drew, but in comments.)